lib/logmatcher.c

```diff
@@ -416,6 +418,7 @@ log_matcher_pcre_re_feed_value(LogMatcherPcreRe *self, LogMessage *msg,
        * make it less appearent, this is not always the case: it does not
        * happen if the new value does not fit the old NVEntry).  */
       log_matcher_pcre_re_save_source_value_to_avoid_clobbering(result);
+      result->source_handles_value_changed = TRUE;
```

Comment:
is it enough that we set this to TRUE at this point? I understand that name-value pairs that are set _prior_ to MESSAGE will be indirect values, which will be resolved once MESSAGE is assigned a new value. But I think we could determine that our input is rewritten by the regexp and in those cases we should never use indirect values.

Reply:
Changed the implementation to check it beforehand.